subprojects/core/src/main/java/org/gradle/api/artifacts/ArtifactView.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.specs.Spec;
+
+import java.util.Map;
+
+/**
+ * A view over the artifacts resolved for this set of dependencies.
+ *
+ * By default, the view returns all files and artifacts, but this can be restricted by component identifier or by attributes.
+ *
+ * @since 3.4
+ */
+@Incubating
+public interface ArtifactView {
```

Comment:
I think to be more consistent we should rename the methods. E.g. we usually do not use the `with` prefix.